demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1526 @@
+                    mqttSessionEstablished = true;
+
+                    /* Check if OTA process was suspended and resume if required. */
+                    if( state == OtaAgentStateSuspended )
+                    {
+                        /* Resume OTA operations. */
+                        OTA_Resume();
+                    }
+                    else
+                    {
+                        /* Send start event to OTA Agent.*/
+                        eventMsg.eventId = OtaAgentEventStart;
+                        OTA_SignalEvent( &eventMsg );
+                    }
+                }
+            }
+
+            if( mqttSessionEstablished == true )
+            {
+                /* Acquire the mqtt mutex lock. */
+                if( pthread_mutex_lock( &mqttMutex ) == 0 )
+                {
+                    /* Loop to receive packet from transport interface. */
+                    mqttStatus = MQTT_ProcessLoop( &mqttContext, 0 );
+
+                    pthread_mutex_unlock( &mqttMutex );
+                }
+                else
+                {
+                    LogError( ( "Failed to acquire mutex to execute process loop"
+                    ",errno=%s",
+                    strerror( errno ) ) );
+                }
+
+                if( mqttStatus == MQTTSuccess )
+                {
+                    /* Get OTA statistics for currently executing job. */
+                    OTA_GetStatistics( &otaStatistics );
+
+                    LogInfo( ( " Received: %u   Queued: %u   Processed: %u   Dropped: %u",
+                               otaStatistics.otaPacketsReceived,
+                               otaStatistics.otaPacketsQueued,
+                               otaStatistics.otaPacketsProcessed,
+                               otaStatistics.otaPacketsDropped ) );
+                }
+                else
+                {
+                    LogError( ( "MQTT_ProcessLoop returned with status = %s.",
+                                MQTT_Status_strerror( mqttStatus ) ) );
+
+                    /* Disconnect from broker and close connection. */
+                    disconnect();
+
+                    /* Set connection flag to false. */
+                    mqttSessionEstablished = false;
+
+                    /* Suspend OTA operations. */
+                    otaRet = OTA_Suspend();
+
+                    if( otaRet != OtaErrNone )
+                    {
+                        LogError( ( "OTA failed to suspend. "
+                                    "StatusCode=%d.", otaRet ) );
+                    }
+                    else
+                    {
+                        suspendTimeout = OTA_SUSPEND_TIMEOUT_MS;
+
+                        while( ( ( state = OTA_GetState() ) != OtaAgentStateSuspended ) && ( suspendTimeout > 0U ) )
+                        {
+                            /* Wait for OTA Library state to suspend */
+                            sleep( OTA_EXAMPLE_TASK_DELAY_MS );
```

Comment:
`sleep` takes a parameter in seconds, but this macro says MS

Reply:
Thanks used `Clock_SleepMs` in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/bc13b469297f8bfa2c27c3738322e60cc4bee1d1